common.mk

```diff
@@ -394,14 +394,14 @@ BASE_LIB_PATH      := $(LIB_PATH)
 LIBBLIS            := libblis
 
 # The shared (dynamic) library file suffix is different for Linux and OS X.
-ifeq ($(OS_NAME),Darwin)
-SHLIB_EXT          := dylib
-else ifeq ($(IS_WIN),yes)
+ifeq ($(IS_WIN),yes)
 ifeq ($(CC_VENDOR),gcc)
 SHLIB_EXT          := dll.a
 else
 SHLIB_EXT          := lib
 endif
+else ifeq ($(OS_NAME),Darwin)
```

Comment:
This is only partially related, but we should do the as `IS_WIN` here. (Set `IS_DARWIN` by checking for __APPLE__ macro)

Reply:
I'm not sure what you're suggesting. `IS_WIN` is checked, I just moved that check before checking `OS_NAME`. I don't think that having `OS_NAME=Darwin` is necessarily bad in this circumstance, and I was able to successfully create a .dll on my machine.

Reply:
I mean when cross compiling for macOS from Linux.

Reply:
Ah. I didn't even know such a thing was possible. I'll save that for another issue/PR.